dblookupext/esdtSupply/logsGetter.go

```diff
@@ -0,0 +1,95 @@
+package esdtSupply
+
+import (
+	"encoding/hex"
+
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+type logsGetter struct {
+	logsStorer  storage.Storer
+	marshalizer marshal.Marshalizer
+}
+
+func newLogsGetter(
+	marshalizer marshal.Marshalizer,
+	logsStorer storage.Storer,
+) *logsGetter {
+	return &logsGetter{
+		logsStorer:  logsStorer,
+		marshalizer: marshalizer,
+	}
+}
+
+func (lg *logsGetter) getLogsBasedOnBody(blockBody data.BodyHandler) (map[string]data.LogHandler, error) {
+	body, ok := blockBody.(*block.Body)
+	if !ok {
+		return nil, errCannotCastToBlockBody
+	}
+
+	logsDB := make(map[string]data.LogHandler)
+	for _, mb := range body.MiniBlocks {
+		shouldIgnore := mb.Type != block.TxBlock && mb.Type != block.SmartContractResultBlock
+		if shouldIgnore {
+			continue
+		}
+
+		dbLogsMb := lg.getLogsBasedOnMB(mb)
+
+		logsDB = mergeLogsMap(logsDB, dbLogsMb)
+	}
+
+	return logsDB, nil
+}
+
+func (lg *logsGetter) getLogsBasedOnMB(mb *block.MiniBlock) map[string]data.LogHandler {
+	dbLogs := make(map[string]data.LogHandler)
+	for _, txHash := range mb.TxHashes {
+		txLog, ok := lg.getTxLog(txHash)
+		if !ok {
+			continue
```

Comment:
by using this mechanism you will just print some warns but fail to correctly process data. This will result in presenting erroneous information. This should be considered processing error and should prevent the advancement of the node

Reply:
I know but in this case, I don't know if a transaction has logs or not.

Reply:
getTxLog function with L72. That is my main concern. Should refactor so the function will return a struct containing the log handler, error and if it contained logs or not so it should be skipped or not on L53